sylvia/tests/reply_data.rs

```diff
@@ -0,0 +1,287 @@
+
+    #[sv::msg(reply, reply_on=success)]
+    fn data_opt(
+        &self,
+        _ctx: ReplyCtx,
+        #[sv::data(opt)] _data: Option<String>,
+        #[sv::payload(raw)] _payload: Binary,
+    ) -> Result<Response, ContractError> {
+        Ok(Response::new())
+    }
+
+    #[sv::msg(reply, reply_on=success)]
+    fn data(
+        &self,
+        _ctx: ReplyCtx,
+        #[sv::data] _data: String,
+        #[sv::payload(raw)] _payload: Binary,
+    ) -> Result<Response, ContractError> {
+        Ok(Response::new())
+    }
+
+    #[sv::msg(reply, reply_on=success)]
+    fn no_data(
+        &self,
+        _ctx: ReplyCtx,
+        #[sv::payload(raw)] _payload: Binary,
+    ) -> Result<Response, ContractError> {
+        Ok(Response::new())
+    }
+}
+
+#[test]
+fn dispatch_replies() {
+    use crate::noop_contract::sv::mt::CodeId as NoopCodeId;
+    use crate::sv::mt::{CodeId, ContractProxy};
+    use crate::sv::{DATA_OPT_REPLY_ID, DATA_RAW_OPT_REPLY_ID, DATA_RAW_REPLY_ID, DATA_REPLY_ID};
+
+    use cosmwasm_std::{to_json_binary, Binary, StdError};
+    use sylvia::cw_multi_test::IntoBech32;
+    use sylvia::multitest::App;
+
+    let app = App::default();
+    let code_id = CodeId::store_code(&app);
+    let noop_code_id = NoopCodeId::store_code(&app);
+
+    let owner = "owner".into_bech32();
+    let data = Some(to_json_binary(&String::from("some_data")).unwrap());
+    let invalid_data = Some(Binary::from("InvalidData".as_bytes()));
+
+    // Trigger remote instantiation reply
+    let contract = code_id
+        .instantiate(noop_code_id.code_id())
+        .with_label("Contract")
+        .call(&owner)
+        .unwrap();
+
+    // Should forward `data` in every case
+    contract
+        .send_message_expecting_data(None, DATA_RAW_OPT_REPLY_ID)
+        .call(&owner)
+        .unwrap();
+
+    contract
+        .send_message_expecting_data(data.clone(), DATA_RAW_OPT_REPLY_ID)
+        .call(&owner)
+        .unwrap();
+
+    // Should forward `data` if `Some` and return error if `None`
+    let err = contract
+        .send_message_expecting_data(None, DATA_RAW_REPLY_ID)
+        .call(&owner)
+        .unwrap_err();
+    assert_eq!(
+        err,
+        StdError::generic_err("Missing reply data field.").into()
+    );
+
+    contract
+        .send_message_expecting_data(data.clone(), DATA_RAW_REPLY_ID)
+        .call(&owner)
+        .unwrap();
+
+    // Should forward deserialized `data` if `Some` or None and return error if deserialization fails
+    contract
+        .send_message_expecting_data(None, DATA_OPT_REPLY_ID)
+        .call(&owner)
+        .unwrap();
+
+    let err = contract
+        .send_message_expecting_data(invalid_data.clone(), DATA_OPT_REPLY_ID)
+        .call(&owner)
+        .unwrap_err();
+    assert_eq!(
+            err,
+            StdError::generic_err("Invalid reply data: SW52YWxpZERhdGE=\nSerde error while deserializing Error parsing into type alloc::string::String: Invalid type").into()
```

Comment:
I see that sylvia prints reply data. What would happen if the msg content size is for e.g. 1MB?